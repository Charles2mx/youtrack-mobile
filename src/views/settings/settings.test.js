import React from 'react';

import Settings from './settings';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import mocks from '../../../test/mocks';

let apiMock;
const getApi = () => apiMock;
const createStoreMock = mocks.createMockStore(getApi);


describe('Settings', () => {
  let stateMock;
  let ownPropsMock;
  let storeMock;

  let wrapper;

  beforeEach(() => {
    stateMock = {
      app: {
        otherAccounts: [],
        isChangingAccount: false
      }
    };
    ownPropsMock = {};
    storeMock = createStoreMock(stateMock, ownPropsMock);
  });


  describe('Render', () => {
    beforeEach(() => {
      wrapper = doShallow();
    });

    it('should match a snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render root component', () => {
      expect(findByTestId('settings')).toHaveLength(1);
    });

    it('should render `Accounts` component', () => {
      expect(findByTestId('settingsAccounts')).toHaveLength(1);
    });

    it('should render footer', () => {
      expect(findByTestId('settingsFooter')).toHaveLength(1);
    });
  });


  function findByTestId(testId) {
    return wrapper && wrapper.find({testID: testId});
  }

  function doShallow(props) {
    return shallow(
      <Settings
        store={storeMock}
        {...props}
      />
    ).shallow();
  }
});

