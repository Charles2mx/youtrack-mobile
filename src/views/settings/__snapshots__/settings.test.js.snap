// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Render should match a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="settings"
>
  <Header
    onRightButtonClick={[Function]}
    title="Settings"
  />
  <View
    style={
      Object {
        "flex": 1,
        "paddingBottom": 24,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
  >
    <Accounts
      isChangingAccount={false}
      onAddAccount={[Function]}
      onChangeAccount={[Function]}
      onClose={[Function]}
      onLogOut={[Function]}
      openDebugView={[Function]}
      otherAccounts={Array []}
      testID="settingsAccounts"
    />
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
      testID="settingsFooter"
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 18,
            "fontWeight": "500",
            "letterSpacing": -0.19,
            "lineHeight": 20,
          }
        }
      >
        YouTrack Mobile
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#FE0082",
              "fontSize": 16,
              "letterSpacing": -0.19,
              "lineHeight": 20,
              "marginBottom": 8,
              "marginTop": 8,
            }
          }
        >
          jetbrains.com/youtrack
        </Text>
      </TouchableOpacity>
      <TouchableWithoutFeedback
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#737577",
              "fontSize": 14,
              "letterSpacing": -0.17,
            }
          }
        >
          0.1 (build 0)
        </Text>
      </TouchableWithoutFeedback>
    </View>
  </View>
</View>
`;
